tests/tests/test_demo_artifact.py

```diff
@@ -71,7 +76,7 @@ def run_demo_script_up(exit_cond=exit_cond):
             logger.info("Started the demo script")
             password = ""
             time.sleep(60)
-            for line in iter(proc.stdout.readline, ""):
+            for line in proc.stdout:
```

Comment:
At least once in the last couple of months, I have seen this test in CI busy in an infinite loop. So thanks for fixing this!